cdap-ui/server/url-helper.js

```diff
@@ -38,16 +38,21 @@ function extractMarketUrls(cdapConfig) {
     return [defaultMarketUrl];
   }
   const marketUrls = cdapConfig['market.base.urls'].split('+');
-  const filteredMarketUrls = marketUrls.filter(element => element !== defaultMarketUrl); 
+  let filteredMarketUrls;
+  if (!defaultMarketUrl) {
```

Comment:
If it is possible for there to be no defaultMarketUrl, could this sometimes cause an issue on line 38?

Reply:
The assumption is if the market.base.urls is not present then the market.base.url is assumed to be present. But you are right there can be corner case when both of them are not present which will render the hub useless. Anyhow I have added additional check.